app/src/main/java/ch/hikemate/app/ui/profile/EditProfileScreen.kt

```diff
@@ -157,15 +197,15 @@ fun EditProfileScreen(
             buttonType = ButtonType.PRIMARY,
             label = context.getString(R.string.edit_profile_screen_save_button_text),
             onClick = {
-              val savedProfile =
+              savedProfile =
                   Profile(
                       profile.id,
                       name,
                       profile.email,
                       HikingLevel.values()[hikingLevel],
                       profile.joinedDate)
-              profileViewModel.updateProfile(profile = savedProfile)
-              navigationActions.goBack()
+              profileViewModel.updateProfile(profile = savedProfile!!)
+              isLoading = true
```

Comment:
Warning: you set it to true here but never to false. What would happen if there is an error while updating the profile? Wouldn't the loading keep going ?

Reply:
Nop since we are supposed to change to another screen after it